Monika After Story/game/cardgames.rpy

```diff
@@ -0,0 +1,5639 @@
+                nou_reminder_timeout - (int) the turn when this player cannot be caught for not saying 'NOU' any longer
+            """
+            def __init__(self, leftie=False):
+                """
+                Constructor
+
+                IN:
+                    leftie - is player leftie or rightie
+                """
+                self.leftie = leftie
+                self.isAI = False
+                self.hand = None
+                self.drew_card = False
+                self.should_draw_cards = 0
+                self.played_card = False
+                self.plays_turn = False
+                self.should_skip_turn = False
+                self.yelled_nou = False
+                self.should_play_card = False
+                self.nou_reminder_timeout = 0
+
+            def __repr__(self):
+                return "<__Player {0}>".format(persistent.playername)
+
+        class __AI(__Player):
+            """
+            AI variation of player
+
+            PROPERTIES:
+                everything from __Player
+                game - (NoU) pointer for internal use
+                cards_data - (dict) data about our cards (amount, values, ids)
+                queued_card - (__Card) the card Monika wants to play on the next turn
+                player_cards_data - (dict) potentially the most common colour (or None)
+                    and most rare colours (or an empty list) in the Player's hand
+                    'reset_in' shows how much turns left until we reset 'has_colour'
+                reactions - (list) all reactions that monika had during this game
+                    (even if they didn't trigger)
+            """
+            MIN_THONK_TIME = 0.3
+            MAX_THONK_TIME = 1.7
+            SHUFFLING_CHANCE = 15
+            LOW_MISSING_NOU_CHANCE = 5
+            HIGHT_MISSING_NOU_CHANCE = 25
+
+            def __init__(self, game, leftie=False):
+                """
+                Constructor
+
+                IN:
+                    leftie - is this player leftie or rightie
+                    game - pointer to our NoU object
+                """
+                super(game.__AI, self).__init__(leftie)
+
+                self.isAI = True
+                self.game = game
+                self.queued_card = None
+                self.chosen_colour = None
+                self.player_cards_data = {
+                    "reset_in": 0,# we reset 'has_colour' to None in 'reset_in' turns
+                    "has_colour": None,# we keep track only on 1 colour here
+                    "lacks_colours": []# but keep track on multiple colours here
+                }
+                # all previous reactions
+                # items format: {"type": type, "turn": turn_id, "monika_card": card or None, "player_card": card or None, "seen_count": int, "shown": boolean}
+                self.reactions = []
+
+            def __repr__(self):
+                return "<__AI Monika>"
```

Comment:
Not sure how useful this repr is given it's static

Reply:
Used it just to distinguish classes during testing, I think it wouldn't hurt if I leave it?

Reply:
It wouldn't hurt, but imo we should make reprs more functional if we're going to use them. Since this would be the same for both the player and monika if both are init'd, this serves no purpose to differentiate, especially since it overrides even the id of it.